File: codecamp/src/main/java/com/gdogaru/codecamp/model/Venue.java
/*
 * Copyright (C) 2008 Gabriel Dogaru (gdogaru@gmail.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.gdogaru.codecamp.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Created by Gabriel Dogaru (gdogaru@gmail.com)
 */
public class Venue implements Serializable {
    @SerializedName("Id")
    long id;
    @SerializedName("name")
    String name;
    @SerializedName("city")
    String city;
    @SerializedName("latitude")
    double latitude;
    @SerializedName("longitude")
    double longitude;
    @SerializedName("country")
    String country;
    @SerializedName("directions")
    String directions;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getDirections() {
        return directions;
    }

    public void setDirections(String directions) {
        this.directions = directions;
    }
}

File: codecamp/src/main/java/com/gdogaru/codecamp/api/model/Venue.kt
/*
 * Copyright (c) 2019 Gabriel Dogaru - gdogaru@gmail.com
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *  *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package com.gdogaru.codecamp.api.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

/**
 * @author Gabriel Dogaru (gdogaru@gmail.com)
 */
data class Venue(
        @JsonProperty("Id")
        var id: Long = 0,

        @JsonProperty("name")
        var name: String,

        @JsonProperty("city")
        var city: String,

        @JsonProperty("latitude")
        var latitude: Double = 0.toDouble(),

        @JsonProperty("longitude")
        var longitude: Double = 0.toDouble(),

        @JsonProperty("country")
        var country: String,

        @JsonProperty("directions")
        var directions: String
) : Serializable
